src/common/base/abstractReportOnPromote.ts

```diff
@@ -72,7 +72,7 @@ export abstract class ReportOnPromoteCommand<T extends typeof SfCommand> extends
       };
     } catch (err) {
       const error = err as Error;
-      if (error.name === 'SingleRecordQuery_NoRecords') {
+      if (error.name === 'No-results-foundError') {
```

Comment:
This seems brittle.  What if another query in this try block fails?  See my comment below.  I think selectOneDeploymentResultByAsyncJobId should be allowed to return null.